plugin-server/src/main/ingestion-queues/session-recordings-consumer.ts

```diff
@@ -42,267 +47,283 @@ export const startSessionRecordingEventsConsumer = async ({
+        other consumers.
+
+        We keep track of unaknowledged messages per Kafka partition, and only
+        commit up to just before the oldest unaknowledged message. This is to
+        ensure that we don't commit offsets for messages that we haven't
+        processed yet. This is important because we use the offset to determine
+        where to pick up processing if the consumer dies and needs restarting.
+        Essentially we want to provide at least once delivery guarantees to the
+        topic we produce to, but we do not currently try to provide exactly once
+        guarantees.
+
+        We try to consumer from Kafka as fast as possible, but we also need to
+        be careful not to consumer too many resources on of the consumer member,
+        so we apply some back pressure based on the number of unaknowledged
+        messages. If the number of unaknowledged messages is greater than
+        `maxUnacknowledgedMessages`, we pause consumption from the partition.
     */
 
-    // We use our own producer as we want to ensure that we don't have out of
-    // band calls to the `flush` method via the KAFKA_FLUSH_FREQUENCY_MS option.
-    // This ensures that we can handle Kafka Producer errors within the body of
-    // the Kafka consumer handler.
     const groupId = 'session-recordings'
     const sessionTimeout = 30000
+    const fetchBatchSize = 500
 
     status.info('🔁', 'Starting session recordings consumer')
 
-    const producer = await createKafkaProducer(kafkaConfig)
-    const consumer = kafka.consumer({
-        groupId: groupId,
-        sessionTimeout: sessionTimeout,
-        maxBytes: consumerMaxBytes,
-        maxBytesPerPartition: consumerMaxBytesPerPartition,
-        maxWaitTimeInMs: consumerMaxWaitMs,
-    })
-    setupEventHandlers(consumer)
-    await consumer.connect()
-    await consumer.subscribe({ topic: KAFKA_SESSION_RECORDING_EVENTS })
-
-    await consumer.run({
-        partitionsConsumedConcurrently,
-        eachBatch: async (payload) => {
-            return await instrumentEachBatch(
-                KAFKA_SESSION_RECORDING_EVENTS,
-                eachBatch({ producer: producer, teamManager, groupId }),
-                payload
-            )
-        },
+    const connectionConfig = createRdConnectionConfigFromEnvVars(kafkaConfig)
+    const producer = await createKafkaProducer(connectionConfig)
+
+    // Create a node-rdkafka consumer.
+    const consumer = await createKafkaConsumer({
+        ...connectionConfig,
+        'group.id': groupId,
+        'session.timeout.ms': sessionTimeout,
+        // Our offset commit strategy is as mentioned on
+        // https://github.com/confluentinc/librdkafka/blob/master/INTRODUCTION.md#at-least-once-processing
+        // i.e. we have librdkafka handle actually committing offsets to Kafka
+        // periodically, but we handle the storing of which offsets we would
+        // like to be committed manually.
+        'enable.auto.commit': true,
```

Comment:
IIRC we need to set this to false if we manually call `consumer.commit`

```suggestion
        'enable.auto.commit': false,
```